### PR TITLE
Add/hashmail email provider

### PR DESCRIPTION
**Pull request by bibaswan7**
### What change does this PR introduce?

It adds hashmail email provider to Novu
API reference: https://docs.hashmail.dev/docs/trigger-notifications-with-api

### Why was this change needed?

Solves #4337 

### Other information (Screenshots)

image

### CODE REVIEW

**Comment by bibaswan7** [comment]
Note: Hashmail doesn't provide any Nodejs SDK

**Comment by davidsoderberg** [comment]
Hi @bibaswan7 please update your branch with latest from next

**Comment by Cliftonz** [comment]
@bibaswan7 can you look at the conflicts?

**Comment by bibaswan7** [comment]
I have just resolved the conflicts @Cliftonz

File: providers/hashmail/src/lib/hashmail.provider.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { HashmailEmailProvider } from './hashmail.provider';
+
+const mockConfig = {
+  apiKey:
+    // eslint-disable-next-line max-len
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiIDogImF1dGhlbnRpY2F0ZWQiLCAiZXhwIiA6IDE3Mjc3Njc0MTYuNzcxMTcwLCAic3ViIiA6ICIzMDA5MjhkNS1jMzZjLTRjYjYtOTMxYS1mYzMwMTViNTdkOGIiLCAicm9sZSIgOiAiYXV0aGVudGljYXRlZCIsICJkZXNjcmlwdGlvbiIgOiAiZGVmYXVsdCIsICJpc3MiIDogImNvbnNvbGUuaGFzaG1haWwuZGV2In0.9POGd1vGHHcFCmMwE3CjMRjISLOjVs-RU0Foejoemeo',
```
[the remark this answers is not in the document]
Comment (bibaswan7) [reply providers/hashmail/src/lib/hashmail.provider.spec.ts:6]: yeah, it has probably expired by now

**Comment by jainpawan21** [comment]
@bibaswan7 

Closing this PR due to inactivity and lot of merge conflicts

**Pull request closed without merging**